snapcraft/projects.py

```diff
@@ -884,6 +900,52 @@ def unmarshal(cls, data: Dict[str, Any]) -> "ArchitectureProject":
         return architectures
 
 
+class ComponentProject(ProjectModel, extra=pydantic.Extra.ignore):
```

Comment:
@lengau suggested we can later have support to handle all early project extractions in a single model.